src/psyclone/domain/common/transformations/alg_invoke_2_psy_call_trans.py

```diff
@@ -130,10 +134,16 @@ def _add_arg(arg, arguments):
         elif isinstance(arg, CodeBlock):
             arguments.append(arg.copy())
         else:
+            if isinstance(arg, Node):
+                string = f"but '{arg.debug_string()}'"
+                if arg.parent and isinstance(arg.parent, KernelFunctor):
+                    string = f"{string} in kernel '{arg.parent.name}'"
+                string = f"{string} is of type '{type(arg).__name__}'."
+            else:
+                string = f"but found '{type(arg).__name__}'."
             raise TypeError(
                 f"Expected Algorithm-layer kernel arguments to be "
-                f"a literal, reference or code block, but "
-                f"found '{type(arg).__name__}'.")
+                f"a literal, reference or code block, {string}.")
```

Comment:
Probably better to capitalise these to emphasise that they are Node types (i.e. "Literal, Reference or CodeBlock").